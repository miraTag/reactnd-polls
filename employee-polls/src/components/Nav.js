import { Link } from "react-router-dom";
import { connect } from "react-redux";
import { handleLogout } from "../actions/authUser_actions";
const Nav = ({ dispatch, authUserId, avatarURL }) => {
  const logout = (e) => {
    e.preventDefault();
    dispatch(handleLogout());
  };
  return (
    <nav className="d-flex align-items-center justify-content-center justify-content-lg-between">
      <div className="text-start">
        <Link to="/" className="px-3 py-2 rounded-lg">
          Home
        </Link>
        <Link to="/leaderboard" className="px-3 py-2 rounded-lg">
          Leaderboard
        </Link>
        <Link to="/new" className="px-3 py-2">
          New Poll
        </Link>
      </div>
      <div className="text-end" data-testid="user-info">
        <img
          src={avatarURL}
          alt={authUserId}
          width="32"
          height="32"
          className="rounded-circle"
          data-testid="user-avatar"
        />
        <span className="px-3 py-2" data-testid="user-id">
          {authUserId}
        </span>
        <button
          onClick={logout}
          className="px-3 py-2 btn-primary btn rounded-lg "
        >
          Logout
        </button>
      </div>
    </nav>
  );
};
const mapStateToProps = ({ authUser: { id, avatarURL } }) => ({
  authUserId: id,
  avatarURL,
});
export default connect(mapStateToProps)(Nav);
