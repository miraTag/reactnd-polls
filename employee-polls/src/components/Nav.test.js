import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "../store";
import { BrowserRouter } from "react-router-dom";
import React from "react";
import Nav from "./Nav";
import { setAuthUser } from "../actions/authUser_actions";

describe("Nav", () => {
  it("should render the component", () => {
    store.dispatch(setAuthUser({ id: "tylermcginnis", password: "" }));

    const component = render(
      <Provider store={store}>
        <BrowserRouter>
          <Nav />
        </BrowserRouter>
      </Provider>
    );
    expect(component).toBeDefined();
    expect(component).toMatchSnapshot();
  });

  it("should display logged in users avatar", () => {
    store.dispatch(setAuthUser({ id: "tylermcginnis", password: "" }));

    const component = render(
      <Provider store={store}>
        <BrowserRouter>
          <Nav />
        </BrowserRouter>
      </Provider>
    );

    const userSpanElement = component.getByTestId("user-info");
    const imgElement = userSpanElement.querySelector("img");

    expect(imgElement).toBeInTheDocument();
  });
});
