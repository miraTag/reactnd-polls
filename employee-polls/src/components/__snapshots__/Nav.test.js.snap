// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should render the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="d-flex align-items-center justify-content-center justify-content-lg-between"
      >
        <div
          class="text-start"
        >
          <a
            class="px-3 py-2 rounded-lg"
            href="/"
          >
            Home
          </a>
          <a
            class="px-3 py-2 rounded-lg"
            href="/leaderboard"
          >
            Leaderboard
          </a>
          <a
            class="px-3 py-2"
            href="/new"
          >
            New Poll
          </a>
        </div>
        <div
          class="text-end"
          data-testid="user-info"
        >
          <img
            alt="tylermcginnis"
            class="rounded-circle"
            data-testid="user-avatar"
            height="32"
            width="32"
          />
          <span
            class="px-3 py-2"
            data-testid="user-id"
          >
            tylermcginnis
          </span>
          <button
            class="px-3 py-2 btn-primary btn rounded-lg "
          >
            Logout
          </button>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="d-flex align-items-center justify-content-center justify-content-lg-between"
    >
      <div
        class="text-start"
      >
        <a
          class="px-3 py-2 rounded-lg"
          href="/"
        >
          Home
        </a>
        <a
          class="px-3 py-2 rounded-lg"
          href="/leaderboard"
        >
          Leaderboard
        </a>
        <a
          class="px-3 py-2"
          href="/new"
        >
          New Poll
        </a>
      </div>
      <div
        class="text-end"
        data-testid="user-info"
      >
        <img
          alt="tylermcginnis"
          class="rounded-circle"
          data-testid="user-avatar"
          height="32"
          width="32"
        />
        <span
          class="px-3 py-2"
          data-testid="user-id"
        >
          tylermcginnis
        </span>
        <button
          class="px-3 py-2 btn-primary btn rounded-lg "
        >
          Logout
        </button>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
